dbt_semantic_interfaces/validations/where_filters.py

```diff
@@ -0,0 +1,268 @@
+                        custom_granularity_names=custom_granularity_names,
+                    )
+
+            denominator = metric.type_params.denominator
+            if denominator is not None and denominator.filter is not None:
+                try:
+                    denominator.filter.filter_expression_parameter_sets
+                except Exception as e:
+                    issues.append(
+                        generate_exception_issue(
+                            what_was_being_done=f"trying to parse the denominator filter on metric `{metric.name}`",
+                            e=e,
+                            context=context,
+                            extras={
+                                "traceback": "".join(traceback.format_tb(e.__traceback__)),
+                            },
+                        )
+                    )
+                else:
+                    issues += WhereFiltersAreParseableRule._validate_time_granularity_names_for_metric(
+                        context=context,
+                        filter_expression_parameter_sets=denominator.filter.filter_expression_parameter_sets,
+                        custom_granularity_names=custom_granularity_names,
+                    )
+
+            for input_metric in metric.type_params.metrics or []:
+                if input_metric.filter is not None:
+                    try:
+                        input_metric.filter.filter_expression_parameter_sets
+                    except Exception as e:
+                        issues.append(
+                            generate_exception_issue(
+                                what_was_being_done=f"trying to parse filter for input metric `{input_metric.name}` "
+                                f"on metric `{metric.name}`",
+                                e=e,
+                                context=context,
+                                extras={
+                                    "traceback": "".join(traceback.format_tb(e.__traceback__)),
+                                },
+                            )
+                        )
+                    else:
+                        issues += WhereFiltersAreParseableRule._validate_time_granularity_names_for_metric(
+                            context=context,
+                            filter_expression_parameter_sets=input_metric.filter.filter_expression_parameter_sets,
+                            custom_granularity_names=custom_granularity_names,
+                        )
+        return issues
+
+    @staticmethod
+    @validate_safely(whats_being_done="running manifest validation ensuring all metric where filters are parseable")
+    def validate_manifest(semantic_manifest: SemanticManifestT) -> Sequence[ValidationIssue]:  # noqa: D
+        issues: List[ValidationIssue] = []
+        custom_granularity_names = [
+            granularity.name
```

Comment:
This is a nit because it's not likely significant to performance, but we could include the standard granularity names here as well so that we don't have to loop through them each time we call `_validate_time_granularity_names_impl()`.

Reply:
Or even better would be to add a cached property using like `WhereFiltersAreParseableRule.valid_granularity_names()`, using `@lru_cache`, to encapsulate this in a nicer way.

Reply:
(I've messaged you to clarify a few things before I respond here :-) )

Reply:
My instinct here is that
1. I do not mind precomputing that here at all
2. I don't think using the LRU is a good idea.  In this case, it's just about code clarity - using a "Least Recently Used Cache" communicates that the contents are incomplete, that we'll be dropping some contents as we add others, etc etc.  That's absolutely not the case here - this is intended to be an exhaustive set, static and useable throughout this entire completely linear run.

Reply:
Do you have a different idea in mind? Or just prefer to keep it as is?

Reply:
Oh, sorry, I thought your first suggestion was to add the standard granularities to the computation right here and just pass it through.  For the purpose of simplicity, that would be my preference.

If we really want to stick this in a variable, I'd prefer to make this an instance-based class instead of using static methods and then instantiate WhereFiltersAreParseableRule with a set of valid_granularities, but that's going to create even more noise in this PR.

Reply:
Yes, I think that was actually my suggestion in my first round of feedback (to make this instance-based)! This is more of a nit though and I'm fine with just adding the standard granularities here and passing them all through.